attiny/attiny.ino

```diff
@@ -35,22 +40,72 @@ enum State {
   PI_WDT_FAILED             // WDT for RPi failed. Turn off and on.
 };
 
+#define I2C_READ_REG_LEN 1
+#define I2C_READ_BATTERY_VOLTAGE 0x20
+
+volatile byte i2cReadRegVal = 0x00;
+volatile byte i2cReadRegs[I2C_READ_REG_LEN] =
+{
+    I2C_READ_BATTERY_VOLTAGE,
+};
+
 State state = PI_POWERED;
 
 void setup() {
+  pinMode(BATTERY_VOLTAGE_PIN, INPUT);
   pinMode(PI_POWER_PIN, OUTPUT);
   pinMode(POWER_LED, OUTPUT);
+  digitalWrite(PI_POWER_PIN, LOW);
+  digitalWrite(POWER_LED, LOW);
+  checkBattery();
   digitalWrite(POWER_LED, LOW);
   initTimer1();
   sei(); // enable interupts
   TinyWireS.begin(I2C_SLAVE_ADDRESS);
   TinyWireS.onReceive(receiveEvent);
   TinyWireS.onRequest(requestEvent);
   setBlinks(2);
-  wdt_enable(WDTO_8S); // enable 8 second WDT
 }
 
-void loop() { 
+void checkBattery() {
+  int a = analogRead(BATTERY_VOLTAGE_PIN);
+  if (425 <= a && a <= 645) {           // Check if battery is between 5.5V and 6.9V
+    setup_watchdog_interrpt();          // Use WDT for waking up device every 8 seconds.
+    digitalWrite(PI_POWER_PIN, LOW);    // Single long LED flash to indicate low battery
+    digitalWrite(POWER_LED, HIGH);
+    delay(1000);
+    digitalWrite(POWER_LED, LOW);
+    while (425 <= a && a <= 696) {      // Check if the battery is still flat.
```

Comment:
given this logic is repeated, perhaps extract this in a `is_battery_low` function?  That would remove the need for the comments.

Reply:
This logic uses a different value for the high value (696 not 645) This is so if the device is being solar charged or small changes in the battery voltage reading won't make it turn on again. 696 is about 7.2V. I'll add a comment for this better explaining it.